src/shared/bg_pmove.cpp

```diff
@@ -168,31 +168,29 @@ static void PM_StartWeaponAnim( int anim )
 PM_StartLegsAnim
 ===================
 */
+// checks for higher priority animation
+// return false if a higher priority animation is running,
+// there was a comment for the segmentedModel case:
+// "legsTimer is clamped too tightly for nonsegmented models"
+// keeping it here in case it makes sense to someone later.
+// FYI: this code was originally 4 branchs.
+static bool PM_LegsPriority( playerState_t const* ps )
```

Comment:
I think you'll change your mind after some commits, since code surrounding `PM_ContinueLegsAnim` and `PM_StartLegsAnim` gets moved to be organised in a more logical way (that is, according if the model is segmented or not). At this point, `PM_LegsPriority` only gets called twice, but I agree that a better name could be found. At least, to have the code in a single place means that any required change would only need to happen at a single place.

Reply:
Yeah, part of my objection is the semantics are unclear from the name. But also it returns a different result from the original code, in the case of a segmented model with torsoTimer > 0.

How about writing the code more concisely in its original spot like

```
int timer = IsSegmentedModel(pm->ps) ? pm->ps->legsTimer : pm->ps->torsoTimer;
if(timer > 0) {
  return; // high preority animation is running
}
```

Reply:
I just reverted the change this commit did, and used the IsSegmentedModel() in the conditions. I think this solves the issue (maybe some more cleanup will come around later, still so much work to do in there...)